### PR TITLE
Add focus area URLs to Preview Pages

### PR DESCRIPTION
**Pull request by otterbotter**
Similar to `preview-pages` except that it excludes JD's changes.

### CODE REVIEW

**Comment by jbothma** [comment]
looks good - next step is to deploy to staging and run a data build against it, right?

**Comment by delenamalan** [comment]
Perhaps just add a check to see if the `focus_areas` value is present in the preview page in the [`test_department`](https://github.com/vulekamali/datamanager/blob/master/budgetportal/tests/test_department.py#L370) tests. Would also be nice if that test could check that the correct `focus_areas` values are added.

**Comment by otterbotter** [comment]
Conflicts have been resolved

**Comment by otterbotter** [comment]
I've added tests for the focus areas, to check if the key is present and that the focus areas are generated correctly.

I also made a change to the way focus areas are included: previously if an empty focus area was found it would be skipped. It now raises an exception if there is an empty focus area.

File: budgetportal/models.py [hunk 1]
```diff
@@ -1417,6 +1421,11 @@ def get_preview_page(self, financial_year_id, phase_slug, government_slug, spher
             programme_ref,
         ]
 
+        # Remove conditional when Actual vs Budgeted Provincial has function_ref available
```
[the remark this answers is not in the document]
Comment (otterbotter) [reply budgetportal/models.py]: I've removed the conditionals and confirmed that the function dimension works with the Actual and Budgeted Provincial Expenditure dataset.

**Pull request merged**